web/backend-fastapi/app/controllers/chat_RAG.py

```diff
@@ -1,41 +1,65 @@
-from fastapi import APIRouter, Depends
-from langchain_community.embeddings import HuggingFaceEmbeddings
-from pydantic import BaseModel
-from typing import List
-from app.dependencies import get_user_info
+from collections import defaultdict
+from fastapi import APIRouter, Body, HTTPException
 from app.models import neo4j, trulens, rag
+from app.rag_states import ImagesAndChunks, AtomicIndexing, UpdatedChunks, StateType
+from ..common.chat_objects import ChatRequest
 
 router = APIRouter()
 kg = None
 tru = None
-embeddings = HuggingFaceEmbeddings(model_name="all-MiniLM-L6-v2")
 
-
-class ChatHistory(BaseModel):
-    prompt: str
-    response: str
-
-
-class ChatRequest(BaseModel):
-    prompt: str
-    chatHistory: List[ChatHistory]
+# Include active states in this state_list
+state_list = [UpdatedChunks, AtomicIndexing, ImagesAndChunks]
+state_map = defaultdict()
+# Each state must be initialized and added to the state_map
+for state in state_list:
+    constructed_state = state()
+    state_map.update(
+        {
+            constructed_state.tag: {
+                "state": constructed_state,
+                "type": constructed_state.type,
+                "trulens_id": constructed_state.trulens_id,
+                "description": constructed_state.description,
+            }
+        }
+    )
 
 
 @router.post("/chat/")
-async def chat(chat_request: ChatRequest):
+async def chat(chat_request: ChatRequest = Body(ChatRequest)):
+    chat_history = []
+    # Ensure the input is a valid dictionary or object
+    if not isinstance(chat_request, ChatRequest):
+        raise ValueError("Input should be a valid ChatRequest object")
     # Global variables initialization
-    global kg, tru, embeddings, APP_ID, session, bedrock_runtime
+    global kg, tru
     rag_fn = rag.get_full_rag()
     if kg is None:
         kg = neo4j.neo4j()
     if tru is None:
         tru = trulens.connect_trulens()
-    tru_rag = trulens.tru_rag(rag_fn)
-    if embeddings is None:
-        embeddings = HuggingFaceEmbeddings(model_name="all-MiniLM-L6-v2")
+
+    state_entry = state_map.get(
+        chat_request.key, state_map[UpdatedChunks().tag]  # Default to UpdatedChunks
+    )
+    if state_entry is None:
+        raise HTTPException(status_code=404, detail="RAG state not found")
```

Comment:
ideally we should have a default state. This would be bad user experiences if it returned nothing.

Reply:
There is a default on line 44. Do you think this default should apply even if the requestor provides an invalid state?
Right now, it applies the default if no state was specified but will return 404 if they specify something that doesn't exist.